[font]: improvements to docs (#31094)
expo/expo: docs/pages/versions/unversioned/sdk/font.mdx

## Configuration in app.json/app.config.js

> The preferred way to add fonts to your app is through the provided [config plugin](/config-plugins/introduction/) if you use config plugins in your project ([EAS Build](/build/introduction) or `npx expo run:[android|ios]`). The plugin allows you to embed font files at build time which is more efficient than using [`loadAsync`](#loadasyncfontfamilyorfontmap-source). Learn more in the [fonts guide](/develop/user-interface/fonts/#with-expo-font-config-plugin).

Reviewer: ```suggestion
The recommended way to add fonts to your app is through `expo-font` [config plugin](/config-plugins/introduction/) if you use config plugins in your project ([EAS Build](/build/introduction) or `npx expo run:[android|ios]`). The plugin allows you to embed font files at build time which is more efficient than using [`loadAsync`](#loadasyncfontfamilyorfontmap-source). See the [Fonts](/develop/user-interface/fonts/#with-expo-font-config-plugin) guide on how to use it.
```

This previous text is the standard approach we take for this section's content to describe config plugin usage. I understand the context this change is about. Let's rephrase to imply config plugin is the "recommended" way (also taking the note added below into account). 

In the last line, let's use guide's proper name "Fonts" instead of generally referencing it when linking to the guide.
